### ISSUE TITLE
MinGW not support and must use GitBash

### ISSUE THREAD

**Issue by FuXiii** [KhronosGroup/KTX-Software#572]
When I build this project I must use `GitBash`  ,otherwise it will output `can not found bash` error.

When I use `MinGW` in `GitBash` to build this project, the `cmake` cmd I input is:
```Cmd
cmake . -B build -G "MinGW Makefiles" -D KTX_FEATURE_LOADTEST_APPS=ON -D KTX_FEATURE_DOC=OFF -D KTX_FEATURE_STATIC_LIBRARY=ON -D CMAKE_EXPORT_COMPILE_COMMANDS=1 -D CMAKE_CXX_COMPILER=clang++ -D CMAKE_C_COMPILER=clang
```

and then run `mingw32-make` cmd then the log:

```Cmd
E:\Lib\KTX-Software-4.0.0\lib\filestream.c:327:38: error: use of undeclared identifier 'S_IFSOCK'
    if (ftype == S_IFIFO || ftype == S_IFSOCK)
                                     ^
1 error generated.
mingw32-make[2]: *** [CMakeFiles\ktx.dir\build.make:281: CMakeFiles/ktx.dir/lib/filestream.c.obj] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:1137: CMakeFiles/ktx.dir/all] Error 2
mingw32-make: *** [Makefile:165: all] Error 2
```

**Comment by MarkCallow** [comment]
I do not have the bandwidth to handle supporting any more platforms or compilers. I will be happy to accept a PR that adds the support and adds matching builds to the CI runs.

You can provide the location of the bash executable you wish to use by defining `BASH_EXECUTABLE` when configuring cmake. You can fix the search so MinGW is included in it by modifying `cmake/modules/FindBash.cmake'.

For `S_IFSOCK` on Windows I see I have the following code

```
#if defined(_MSC_VER)
  #define S_IFIFO _S_IFIFO
  #define S_IFSOCK 0xC000
#endif
```

Possibly _MSC_VER needs to be changed to WINDOWS if MingGW is using Windows' headers rather than gcc's.

**Comment by MarkCallow** [comment]
There is a related issue, #328. The OP never provided a PR so it remains open.

**Comment by FuXiii** [comment]
What is OP and RP?

**Comment by MarkCallow** [comment]
OP is "original poster" - the person who created the issue, #328 in this case.

"PR" is "pull request". A pull request is the way people who are not maintainers of a project on GitHub can contribute code to that project. See GitHub's [_About pull requests_](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests).

**Comment by FuXiii** [comment]
Maybe I wish contribute some code for KTX. I was familiar with Vulkan.

**Comment by MarkCallow** [comment]
Fixed by PR #574.
